Scripts/assignment/emme_assignment.py

```diff
@@ -69,14 +71,17 @@ def assign(self, time_period, matrices, is_last_iteration=False, is_first_iterat
         self.emme_project.logger.info("Assignment starts...")
         self.set_emmebank_matrices(matrices)
         scen_id = self.emme_scenarios[time_period]
-        if not self._has_assigned_bike_and_walk:
+        if is_first_iteration:
+            self._prepare_network()
             self._assign_pedestrians(scen_id)
             self._assign_bikes(self.bike_scenario,
                             self.result_mtx["dist"]["bike"]["id"],
                             "all",
                             "@bike_"+time_period)
```

Comment:
So we will run pedestrian and bike assignment three times, even if the results are the same every time. If this is acceptable when it comes to run time, then ok.

Reply:
Something I didn't think about.. Good that you pointed it out! Checked from the log files that pedestrian and bike assignment take less than 10 seconds to complete.

I'll keep this change for now and take this into account when trying to optimize the run time as a whole.